content/blog/tidyclust-0-1-0/index.Rmd

```diff
@@ -0,0 +1,150 @@
+
+```{r}
+kmeans_spec <- k_means(num_clusters = 4) %>%
+  set_engine("ClusterR")
+kmeans_spec
+```
+
+We can use the `set_engine()`, `set_mode()`, and `set_args()` we are familiar with from parsnip. The specification itself isn't worth much if we don't apply it to some data. We will use the ames data set from the modeldata package
+
+```{r}
+data("ames", package = "modeldata")
+```
+
+this data set contains a number of categorical variables that unaltered can't be used with a k-means model. Some light preprocessing can be done using the recipes package.
+
+```{r}
+rec_spec <- recipe(~ ., data = ames) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  step_zv(all_predictors()) %>%
+  step_normalize(all_numeric_predictors()) %>%
+  step_pca(all_numeric_predictors(), threshold = 0.8)
+```
+
+This recipe ensures that all the variables are numeric, normalized and have PCA applied to them to remove some of the correlation and number of features. Notice how we didn't specify an outcome as clustering models are unsupervised, meaning that we don't have outcomes.
+
+These two specifications can be combined in a `workflow()` 
+
+```{r}
+kmeans_wf <- workflow(rec_spec, kmeans_spec)
+```
+
+And fit them together
+
+```{r}
+kmeans_fit <- fit(kmeans_wf, data = ames)
+kmeans_fit
+```
+
+We have arbitrarily set the number of clusters to 4 above. But if we wanted to figure out what values would be "optimal" we would have to fit multiple models. We can do this with `tune_cluster()`, for this we need to use `tune()` to specify that `num_clusters` is the argument we want to try with multiple values
+
+```{r}
+kmeans_spec <- kmeans_spec %>% 
+  set_args(num_clusters = tune())
+
+kmeans_wf <- workflow(rec_spec, kmeans_spec)
+kmeans_wf
+```
+
+Using bootstraps to fit multiple models for each value of `num_clusters` and we can use `tune_cluster` much the same way we use `tune_grid()`
+
+```{r}
+set.seed(1234)
+boots <- bootstraps(ames, times = 10)
+
+tune_res <- tune_cluster(
+  kmeans_wf,
+  resamples = boots
+)
+```
+
+and the respective `collect_*()` functions work as with tune output
+
+```{r}
+collect_metrics(tune_res)
+```
+
+## Extraction
+
+Going back to the first model with fit, in tidyclust there are 3 main things we can do with a fitted cluster model:
+
+- extract cluster assignments
+- extract centroid locations
+- prediction with new data
+
+Each of these tasks has a function associated with them. First, we have `extract_cluster_assignment()`, which can be used on fitted tidyclust objects, alone or as a part of a workflow and it returns the cluster assignment as a factor named `.cluster`in a tibble.
+
+```{r}
+extract_cluster_assignment(kmeans_fit)
+```
+
+The location of the clusters can be found using `extract_centroids()` which again returns a tibble, with `.cluster` being a factor with the same levels as earlier.
```

Comment:
Does "earlier" mean in the structure from the training data?

Reply:
good note! gonna clarify